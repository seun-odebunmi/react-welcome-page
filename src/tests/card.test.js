import React from 'react';
import Card from "../components/card";
import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import Enzyme,{ shallow } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

test("Matches", () => {
    const component = renderer.create(
        <Card
            icon="make-payment.png"
            title="This is not a card"
            desc="This is a card"
            onClick={() => {}} />
    );

    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test("Renders <HR /> & <div>", () => {
    const wrapper = shallow(
        <Card
            icon="make-payment.png"
            title="This is not a card"
            desc="This is a card"
            onClick={() => {}} />
    )

    expect(wrapper.find("hr")).toHaveLength(2);
    expect(wrapper.find("div")).toHaveLength(2);
})