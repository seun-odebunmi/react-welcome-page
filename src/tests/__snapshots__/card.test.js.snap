// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches 1`] = `
<div
  className="card bg-white ba br4 b--black-20 pointer grow"
  onClick={[Function]}
>
  <hr
    className="ma0 ntPoint5 center w-30 ba b--black-20"
  />
  <hr
    className="ma0 ntPoint5 center w-30 ba b--black-20"
  />
  <div
    className="card-body content-center tc pa4 pb5"
  >
    <img
      alt="card-icon"
      className="w-60"
      src="make-payment.png"
    />
    <h2
      className="b f2 mb4"
    >
      This is not a card
    </h2>
    <p
      className="f4 lh-copy"
    >
      This is a card
    </p>
  </div>
</div>
`;
